java/src/main/native/src/CuFileJni.cpp

```diff
@@ -181,7 +183,8 @@ class cufile_file {
    * @return std::unique_ptr<cufile_file> for writing.
    */
   static auto make_writer(char const *path) {
-    auto const file_descriptor = open(path, O_CREAT | O_WRONLY | O_DIRECT, S_IRUSR | S_IWUSR);
+    // NOTE: 0664 is what Spark disk block manager uses.
+    auto const file_descriptor = open(path, O_CREAT | O_WRONLY | O_DIRECT, 0664);
```

Comment:
Do we need this change?  Not sure why we want spilled and intermediate data to be readable by a different user.

Reply:
Just trying to be consistent with what Spark's disk manager is doing.

Reply:
I suspect this is a result of the umask setting rather than what Spark is doing explicitly.  In that case we should be setting this to `0666` and let the user's umask decide the final permissions.

Reply:
I think you are right. The default umask on ubuntu is 002. Reverted this change.